docs/examples/metrics/plot_lcss_custom_metric.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+"""
+LCSS computation with a custom distance metric
+=============================================
+.. currentmodule:: tslearn.metrics
+
+This example illustrates how to use the LCSS computation of the
+alignment path [1]_ on an user-defined distance matrix using
+:func:`dtw_path_from_metric`.
+
+Left is the LCSS of two angular time series using the length of the arc on the
+unit circle as a distance metric [2]_ and right is the LCSS of two
+multidimensional boolean time series using hamming distance [3]_.
+
+The images represent cost matrices, that is, on the left the length of the arc
+between each pair of angles on the unit circle and on the right the hamming
+distances between the multidimensional boolean arrays. In both cases, the
+corresponding time series are represented at the left and at the top of each
+cost matrix.
+
+The alignment path, that is the path where the matches between the two time-series
+occurred within the pre-defined threshold, is represented in white on the image.
+
+.. [1] M. Vlachos, D. Gunopoulos, and G. Kollios. 2002. "Discovering Similar
+    Multidimensional Trajectories", In Proceedings of the 18th International
+    Conference on Data Engineering (ICDE '02). IEEE Computer Society, USA, 673.
+
+.. [2] Definition of the length of an arc on `Wikipedia
+       <https://en.wikipedia.org/wiki/Arc_(geometry)#Length_of_an_arc_of_a_circle>`_.
+
+.. [3] See Hammig distance in `Scipy's documentation
+       <https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.hamming.html>`_.
+
+"""
+
+# Author: Daniela Duarte
+# License: BSD 3 clause
+# sphinx_gallery_thumbnail_number = 2
+
+import numpy as np
+from numpy import pi
+from sklearn.metrics import pairwise_distances
+import matplotlib.pyplot as plt
+from matplotlib.colors import LinearSegmentedColormap
+
+from tslearn.generators import random_walks
+from tslearn import metrics
+from tslearn.preprocessing import TimeSeriesScalerMeanVariance
+
+
+np.random.seed(0)
+n_ts, sz = 2, 100
+
+
+# Example 1 : Length of the arc between two angles on a circle
+def arc_length(angle_1, angle_2, r=1.):
+    """Length of the arc between two angles (in rad) on a circle of
+    radius r.
+    """
+    # Compute the angle between the two inputs between 0 and 2*pi.
+    theta = np.mod(angle_2 - angle_1, 2*pi)
+    if theta > pi:
+        theta = theta - 2 * pi
+    # Return the length of the arc
+    L = r * np.abs(theta)
+    return(L)
+
+
+dataset_1 = random_walks(n_ts=n_ts, sz=sz, d=1)
+scaler = TimeSeriesScalerMeanVariance(mu=0., std=pi)  # Rescale the time series
+dataset_scaled_1 = scaler.fit_transform(dataset_1)
+
+# LCSS using a function as the metric argument
+path_1, sim_1 = metrics.lcss_path_from_metric(
+    dataset_scaled_1[0], dataset_scaled_1[1], metric=arc_length
+)
+
+# Example 2 : Hamming distance between 2 multi-dimensional boolean time series
```

Comment:
The output of this second is not very satisfactory: the diagonal match is fully preserved. Maybe you could change the value of epsilon or completely discard this one, I don't know.

Reply:
second *example*

Reply:
If the diagonal match is preserved, then increasing the epsilon value will have no change. Should I delete it?

Reply:
OK for me.